pkg/osquery/tables/airport/table_darwin_test.go

```diff
@@ -0,0 +1,342 @@
+		executorReturnFilePaths []string
+
+		// path to to the file that is a json of the expected output
+		expectedResultsFilePath string
+	}{
+		{
+			sqlEquivalent:           "select * from kolide_airport_util where option = 'scan'",
+			options:                 []string{"scan"},
+			executorReturnFilePaths: []string{"testdata/scan.input.txt"},
+			expectedResultsFilePath: "testdata/scan.output.json",
+		},
+		{
+			sqlEquivalent:           "select * from kolide_airport_util where option = 'scan' and query = '/SSID'",
+			options:                 []string{"scan"},
+			query:                   "/SSID",
+			executorReturnFilePaths: []string{"testdata/scan.input.txt"},
+			expectedResultsFilePath: "testdata/scan_with_query.output.json",
+		},
+		{
+			sqlEquivalent:           "select * from kolide_airport_util where option = 'getinfo'",
+			options:                 []string{"getinfo"},
+			executorReturnFilePaths: []string{"testdata/getinfo.input.txt"},
+			expectedResultsFilePath: "testdata/getinfo.output.json",
+		},
+		{
+			sqlEquivalent:           "select * from kolide_airport_util where option in ('getinfo', 'scan')",
+			options:                 []string{"getinfo", "scan"},
+			executorReturnFilePaths: []string{"testdata/getinfo.input.txt", "testdata/scan.input.txt"},
+			expectedResultsFilePath: "testdata/getinfo_and_scan.output.json",
+		},
+		{
+			sqlEquivalent:           "select * from kolide_airport_util where option in ('getinfo', 'scan') and query = '/SSID'",
+			options:                 []string{"getinfo", "scan"},
+			query:                   "/SSID",
+			executorReturnFilePaths: []string{"testdata/getinfo.input.txt", "testdata/scan.input.txt"},
+			expectedResultsFilePath: "testdata/getinfo_and_scan_with_query.output.json",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.sqlEquivalent, func(t *testing.T) {
+			t.Parallel()
+
+			queryContext := table.QueryContext{
+				Constraints: map[string]table.ConstraintList{},
+			}
+
+			// add the options constraints
+			if tt.options != nil {
+				optionConstraints := []table.Constraint{}
+
+				for _, option := range tt.options {
+					optionConstraints = append(optionConstraints, table.Constraint{Operator: table.OperatorEquals, Expression: option})
+				}
+
+				queryContext.Constraints["option"] = table.ConstraintList{
+					Affinity:    "TEXT",
+					Constraints: optionConstraints,
+				}
+			}
+
+			// add the query constraints
+			if tt.query != "" {
+				queryContext.Constraints["query"] = table.ConstraintList{
+					Affinity:    "TEXT",
+					Constraints: []table.Constraint{{Operator: table.OperatorEquals, Expression: tt.query}},
+				}
+			}
+
+			executor := &mocks.Executor{}
+
+			for index, inputFile := range tt.executorReturnFilePaths {
+				inputBytes, err := os.ReadFile(inputFile)
+				require.NoError(t, err)
+				executor.On("Exec", tt.options[index]).Return(inputBytes, nil).Once()
```

Comment:
The `index` linkage here feels a little fragile. But maybe not worth digging into a better idiom. May as well just roll with it.

Reply:
converted this to a map where the key is the option and the value is the path to the data